api/inc/register_gateway.h

```diff
@@ -143,77 +145,95 @@
 /** Get the selected bits at the target address.
  * @param box_name[in] Box name as defined by the uVisor box configuration
  *                     macro `UVISOR_BOX_CONFIG`
+ * @param shared[in]   Whether the gateway can be shared with other boxes or
+ *                     not. Two values are available: UVISOR_RGW_SHARED,
+ *                     UVISOR_RGW_EXCLUSIVE.
  * @param address[in]  Target address
  * @param mask[in]     Bits to select out of the target address
  * @returns The value `*address & mask`.
  */
-#define UVISOR_BITS_GET(box_name, address, mask) \
+#define UVISOR_BITS_GET(box_name, shared, address, mask) \
     /* Register gateway implementation:
      * *address & mask */ \
-    uvisor_read(box_name, address, UVISOR_RGW_OP_READ_AND, false, mask)
+    uvisor_read(box_name, shared, address, UVISOR_RGW_OP_READ_AND, mask)
 
 /** Check the selected bits at the target address.
  * @param box_name[in] Box name as defined by the uVisor box configuration
  *                     macro `UVISOR_BOX_CONFIG`
+ * @param shared[in]   Whether the gateway can be shared with other boxes or
+ *                     not. Two values are available: UVISOR_RGW_SHARED,
+ *                     UVISOR_RGW_EXCLUSIVE.
  * @param address[in]  Address at which to check the bits
  * @param mask[in]     Bits to select out of the target address
  * @returns The value `(bool) (*address & mask) == mask)`.
  */
-#define UVISOR_BITS_CHECK(box_name, address, mask) \
-    ((bool) (UVISOR_BITS_GET(box_name, address, mask) == mask))
+#define UVISOR_BITS_CHECK(box_name, shared, address, mask) \
+    ((bool) (UVISOR_BITS_GET(box_name, shared, address, mask) == mask))
```

Comment:
Is `bool` acceptable here? The user would have to have `#include <stdbool.h>` somewhere, but we cannot guarantee this, unless we include it here as well.